src/lib/services/access-service.ts

```diff
@@ -81,95 +93,132 @@ export class AccessService {
 
     private permissions: IPermission[];
 
-    constructor({ accessStore, userStore }: IStores, { getLogger } : { getLogger: Function}) {
+    constructor(
+        { accessStore, userStore }: IStores,
+        { getLogger }: { getLogger: Function },
+    ) {
         this.store = accessStore;
         this.userStore = userStore;
         this.logger = getLogger('/services/access-service.ts');
-        this.permissions = Object.values(p).map(p => ({
+        this.permissions = Object.values(permissions).map(p => ({
             name: p,
-            type: isProjectPermission(p) ? PermissionType.project : PermissionType.root
-        }))
+            type: isProjectPermission(p)
+                ? PermissionType.project
+                : PermissionType.root,
+        }));
     }
 
     /**
      * Used to check if a user has access to the requested resource
-     * 
-     * @param user 
-     * @param permission 
-     * @param projectId 
+     *
+     * @param user
+     * @param permission
+     * @param projectId
      */
-    async hasPermission(user: User, permission: string, projectId?: string): Promise<boolean> {
-        this.logger.info(`Checking permission=${permission}, userId=${user.id} projectId=${projectId}`);
+    async hasPermission(
+        user: User,
+        permission: string,
+        projectId?: string,
+    ): Promise<boolean> {
+        this.logger.info(
+            `Checking permission=${permission}, userId=${user.id} projectId=${projectId}`,
+        );
 
-        const permissions = await this.store.getPermissionsForUser(user.id);
+        const userP = await this.store.getPermissionsForUser(user.id);
 
-        return permissions
-                .filter(p => !p.project || p.project === projectId || p.project === ALL_PROJECTS)
-                .some(p => p.permission === permission || p.permission === ADMIN);
+        return userP
+            .filter(
+                p =>
+                    !p.project ||
```

Comment:
Shouldn't this be &&? If p.project does not exist we don't want to do the following checks.